File: dev/test_data/cpp/test_main/aas_core_meta.v3/input/snippets/verification/extension_names_are_unique.cpp
bool ExtensionNamesAreUnique(
  const std::vector<
    std::shared_ptr<types::IExtension>
  >& extensions
) {
  std::set<std::wstring> name_set;

  for (const std::shared_ptr<types::IExtension>& extension : extensions) {
    const std::wstring& name = extension->name();

    if (name_set.find(name) != name_set.end()) {
      return false;
    }

    name_set.insert(name);
  }

  return true;
}

File: dev/test_data/typescript/test_main/aas_core_meta.v3/input/snippets/Verification/extension_names_are_unique.ts
/**
 * Check that all {@link types.Extension.name} are unique
 * among `extensions`.
 *
 * @param extensions - to be verified
 * @returns `true` if the check passes
 */
export function extensionNamesAreUnique(
  extensions: Iterable<AasTypes.Extension>
): boolean {
  const nameSet = new Set<string>();
  for (const extension of extensions) {
    if (nameSet.has(extension.name)) {
      return false;
    }

    nameSet.add(extension.name);
  }

  return true;
}
